com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -99,129 +86,36 @@ protected String getStepErrorMessageAdditionalDescription(DelegateExecution cont
         return ExceptionMessageTailMapper.map(configuration, CloudComponents.SERVICE_BROKERS, offering);
     }
 
-    private List<CloudServiceExtended> getServicesData(List<String> serviceNames, ExecutionWrapper execution) {
+    private List<CloudServiceInstance> getServiceInstancesData(List<String> servicesToDelete, ExecutionWrapper execution) {
         CloudControllerClient client = execution.getControllerClient();
 
-        return serviceNames.parallelStream()
-                           .map(name -> client.getService(name, false))
-                           .filter(Objects::nonNull)
-                           .map(this::buildCloudServiceExtended)
-                           .collect(Collectors.toList());
-    }
-
-    private ImmutableCloudServiceExtended buildCloudServiceExtended(CloudService service) {
-        return ImmutableCloudServiceExtended.builder()
-                                            .metadata(service.getMetadata())
-                                            .name(service.getName())
-                                            .build();
+        return servicesToDelete.parallelStream()

```

Comment:
This is not really properly parallel. It will have a maximum of 2 threads operating in parallel. You should use ForkJoinPoolUtil to ensure proper parallelism.